### PR TITLE
Added per field k_size to control extend of verification on per field…

### PR DESCRIPTION
**Pull request by muellch**
## Technical Description

This PR addresses the problem that in fused stencils the verification is always done for the full range in k-direction, even if some fields do not exist on all k levels (typically some fields exist from [0, nlev) and some [0, nlevpq)). With this PR one can specify the verification k-range on a by-field basis with optional argument in the fortran interface of the setup method.

### Resolves / Enhances

https://gitlab.dkrz.de/dsl/icon-cscs/-/issues/1
(e.g resolves #XXX)

### Example

The followin fusion four dusk files into one:
```
-    call setup_mo_solve_nonhydro_stencil_03(p_patch(jg)%dsl_gpu_trimesh, p_patch(jg)%nlev, streamid)
-    call setup_mo_solve_nonhydro_stencil_04(p_patch(jg)%dsl_gpu_trimesh, p_patch(jg)%nlevp1, streamid)
-    call setup_mo_solve_nonhydro_stencil_05(p_patch(jg)%dsl_gpu_trimesh, p_patch(jg)%nlev, streamid)
-    call setup_mo_solve_nonhydro_stencil_06(p_patch(jg)%dsl_gpu_trimesh, p_patch(jg)%nlev, streamid)
+    call setup_fused_mo_solve_nonhydro_stencils_03_04_05_06(p_patch(jg)%dsl_gpu_trimesh, p_patch(jg)%nlevp1, streamid)
```
results in an out of bounds error during the verification. Because only stencil 04 is properly defined for upper k level nlevp1. The verification then accesses out of bounds for the other fields (for example for stencil_06) causing a GPUAssert to trigger.

### Testing

https://gitlab.dkrz.de/dsl/icon-cscs/-/merge_requests/45

### Dependencies

None

### CODE REVIEW

**Comment by muellch** [comment]
launch jenkins

File: dawn/src/dawn/CodeGen/Cuda-ico/CudaIcoCodeGen.cpp
Comment (mroethlin) [inline dawn/src/dawn/CodeGen/Cuda-ico/CudaIcoCodeGen.cpp:1755]: camel case

File: dawn/src/dawn/CodeGen/Cuda-ico/CudaIcoCodeGen.cpp
Comment (mroethlin) [inline dawn/src/dawn/CodeGen/Cuda-ico/CudaIcoCodeGen.cpp:1780]: If we were to keep this I'd suggest we either have an auto format hook somewhere in the build pipeline or an indentation stack kinda deal in the FORTRAN interface. For now I'd just extract the indent in a variable like `const std::string indent = "     "`, then subsequently use that

**Review by mroethlin — APPROVED** [review]
Two very small and very nit-picky things that don't need further approval. I'd like to make sure that your changes do not break `icondusk-e2e`. From my understanding they shouldn't, but I still would like to make sure. Currently CI breaks over there with ridiculous errors though. Good to go after green `icondusk-e2e` CI.

**Comment by mroethlin** [comment]
Does not mess with `icondusk-e2e` as confirmed [here](http://localhost:8081/view/ICON%20DSL/job/icondusk-e2e_PR/slave=tsa/138/console). gtg after the small changes!

**Comment by muellch** [comment]
launch jenkins

**Pull request merged**
